### PR TITLE
Add a new panel for all OpenShift commands

### LINKED ISSUE #2819: Create a new Terminal Column for OpenShift Output
There should be a dedicated terminal to show the OpenShift Output commands and should be labelled specificially.

Screenshot 2023-04-05 at 6 46 21 PM

### PR DESCRIPTION
**Pull request by datho7561**
It's a webview, since we need to use either a treeview or webview for all panels. It wraps xtermjs, the same library VS Code uses for its terminal. It exposes an API for interacting with the terminal multiplexer, as well as with the individual running processes. For example, we can run `odo dev` and update the UI to show the state of the deployment created by odo (like we were doing before this PR).

What's missing and can probably saved for a future PR:
- Copy/paste using `Ctrl+Shift+C`/`Ctrl+Shift+V` (using the context menu to copy and paste works)
- Tests (pending on https://github.com/redhat-developer/vscode-extension-tester/pull/855)
- Reordering the tabs (I think we should gauge interest on this feature, since it would be hard to implement)
- Tab bar scrolling when there are too many tabs. Supposedly the Material UI component comes with this built it, but it wasn't working when I tested it

Known bugs:
- If you create two terminal tabs (through commands or running `odo dev`), and switch back to the first tab, the font size and family changes on that tab. All subsequent tab creation/switching doesn't seem to cause this issue.

Closes #2819

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

**Comment by datho7561** [comment]
blocked on https://github.com/redhat-developer/vscode-extension-tester/issues/804

**Comment by datho7561** [comment]
I just realized that, after this change, it will be very difficult to write automated tests for anything that uses the terminal. We are testing using Selenium, which can scrape data off the HTML DOM. The library I am using to render the terminal output uses WebGL for performance reasons, so the terminal text is not in a DOM node, it's a rendered image.

The two ways I can think of to test the terminal output:
- Take a screenshot using selenium and compare the screenshot to a cached screenshot, which is really flaky
- Switch the rendering strategy of the terminal to use a DOM-based version during the test suite, and trust that the renderers are similar enough

I'll try the second version and see if I get anywhere

**Comment by datho7561** [comment]
Requires https://github.com/redhat-developer/vscode-extension-tester/pull/855 (I have been using `npm link` locally)

**Review by msivasubramaniaan — COMMENTED** [review]
The below error on `npm install`

`ERR! find VSfind VS msvs_version not set from command line or npm config?`

After resolved the above error and opened the extension, I was trying to run the `Start Dev` command, then the extension was stuck and I couldn't able to perform anything further on the extension.

**Comment by datho7561** [comment]
`npm i` works for me on Windows, and the webview loads, but when running a command, the output is not transferred into the webview properly. @JessicaJHee or @msivasubramaniaan , can you confirm that the terminal works on Linux?

**Review by JessicaJHee — COMMENTED** [review]
Works great now! The only thing I noticed was that if I tried to close the middle tab, it will close the tabs to the right of it as well. 

Peek 2023-06-29 14-28

**Comment by datho7561** [comment]
> if I tried to close the middle tab, it will close the tabs to the right of it as well.

Should be fixed

File: src/webview/openshift-terminal/app/terminalMultiplexer.tsx
Comment (msivasubramaniaan) [inline src/webview/openshift-terminal/app/terminalMultiplexer.tsx:6]: There were no dependency `'@emotion/react'` in package.json, How it compile this one?

For me the below error during compile:
 ```
*  Executing task: npm run watch 


> vscode-openshift-connector@1.6.0 watch
> node ./build/esbuild-watch.mjs

[1:57:50 AM] File change detected. Compiling...
X [ERROR] Could not resolve "@emotion/react"

    src/webview/openshift-terminal/app/terminalMultiplexer.tsx:6:30:
      6 │ import { ThemeProvider } from '@emotion/react';
        ╵                               ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

X [ERROR] Could not resolve "@emotion/styled"

    node_modules/@mui/styled-engine/index.js:9:21:
      9 │ import emStyled from '@emotion/styled';
        ╵                      ~~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will
  remove this error.

X [ERROR] Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/index.js:34:45:
      34 │ export { ThemeContext, keyframes, css } from '@emotion/react';
         ╵                                              ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

X [ERROR] Could not resolve "@emotion/styled"X [ERROR] Could not resolve "@emotion/styled"

    node_modules/@mui/styled-engine/index.js:9:21:
      9 │ import emStyled from '@emotion/styled';
        ╵

    node_modules/@mui/styled-engine/index.js:9:21:
      9 │ import emStyled from '@emotion/styled';
        ╵                      ~~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will
  remove this error.

X [ERROR]X [ERROR] Could not resolve "@emotion/styled"

    node_modules/@mui/styled-engine/index.js:9:21:
      9 │ import emStyled from '@emotion/styled';
        ╵                      ~~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will
  remove this error.

X [ERROR] Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/index.js:34:45:
      34 │ export { ThemeContext, keyframes, css } from '@emotion/react'X [ERROR]~~~~~~~~~~~~~~~~~ Could not resolve "@emotion/react"

  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will
  remove this error.

 X [ERROR] Could not resolve "@emotion/styled"

    node_modules/@mui/styled-engine/index.js:9:21:
      9 │ import emStyled from '@emotion/styled';
        ╵                      ~~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will
  remove this error.

X [X [
                                                                                                                                                
    node_modules/@mui/styled-engine/StyledEngineProvider/StyledEngineProvider.js:3:30:                                                          
      3 │ import { CacheProvider } from '@emotion/react';
        ╵                               ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

Could not resolve "@emotion/react"ERROR

    node_modules/@mui/styled-engine/index.js:34:45:
]       34 │ export { ThemeContext, keyframes, css } from Could not resolve "@emotion/react"'@emotion/react';
         ╵                                              ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

X [ERROR
                                                                                                                                                
    node_modules/@mui/styled-engine/index.js:34:45:                                                                                             
      34 │ export { ThemeContext, keyframes, css } from ;
         ╵                                              ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

ERROR] Could not resolve "@emotion/styled"

    node_modules/@mui/styled-engine/index.js:9:21:
]      9 │ import emStyled from '@emotion/styled';
        ╵                      ~~~~~~~~~~~~~~~~~X

  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will
  remove this error.

[X ERROR[ERROR]]  X Could not resolve "@emotion/react"[ERROR
                                                                                                                                                
    node_modules/@mui/styled-engine/index.js:34:45:                                                                                             
Could not resolve "@emotion/styled"                                                                                                             

    node_modules/@mui/styled-engine/index.js:9:21:
      9 │ import emStyled from '@emotion/styled';
        ╵                      ~~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will
  remove this error.

X [ERROR] Could not resolve "@emotion/styled"

    node_modules/@mui/styled-engine/node/index.js:46:45:
      46 │ var _styled = _interopRequireDefault(require("@emotion/styled"X [));
         ╵                                              ~~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

'@emotion/react';
         ╵                                              ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

X [ERROR] Could not resolve "@emotion/styled"

    node_modules/@mui/styled-engine/node/index.js:46:45:
      46 │ var _styled = _interopRequireDefault(require("@emotion/styled"));
         ╵                                              ~~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

X [ERROR] Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/index.js:34:45:
      34 │ export { ThemeContext, keyframes, css } from X ['@emotion/react';
         ╵                                              ERROR] ERROR]  Could not resolve "@emotion/react"~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

      34 │ export { ThemeContext, keyframes, css } from '@emotion/react';
         ╵                                              ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

X [ERROR] Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/GlobalStyles/GlobalStyles.js:3:23:
      3 │ import { Global } from '@emotion/react';
        ╵                        ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

X [ERRORX ] Could not resolve "@emotion/styled"

    node_modules/@mui/styled-engine/index.js:9:21:
      9 │ import emStyled from '@emotion/styled';
        ╵                      ~~~~~~~~~~~~~~~~~X

  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will
  remove this error.


                                                                                                                                                
    node_modules/@mui/styled-engine/node/index.js:47:21:                                                                                        
      47 │ var _react = require("@emotion/react");
         ╵                      ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/GlobalStyles/GlobalStyles.js:3:23:
[ERROR] Could not resolve "@emotion/styled"      3 │ import { Global } from '@emotion/react';
        ╵                        X X [X ERROR[]ERROR ]Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/StyledEngineProvider/StyledEngineProvider.js:3:30:
Could not resolve "@emotion/react"      3 │ import { CacheProvider } from

    node_modules/@mui/styled-engine/index.js:34:45:
      34 │ export { ThemeContext, keyframes, css } from '@emotion/react';
         ╵                                              ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

[ERROR] Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/GlobalStyles/GlobalStyles.js:3:23:
[X [ERROR'@emotion/react'

    node_modules/@mui/styled-engine/node/index.js:46:45:
] Could not resolve "@emotion/react"Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/GlobalStyles/GlobalStyles.js:3:23:
]

    node_modules/@mui/styled-engine/StyledEngineProvider/StyledEngineProvider.js:3:30:
      3 │ import { CacheProvider } from '@emotion/react';
        ╵                               ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

X [ERROR] Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/GlobalStyles/GlobalStyles.js:3:23:
      3 │ import { Global } from '@emotion/react'X       3 │ import { Global } from [ERROR];
        ╵                               ;
        ╵                              3 │ import { Global } from '@emotion/react';
        ╵                        ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

ERROR[1:57:54 AM] Compilation complete. Found some errors. Watching for file changes.
       46 │ var _styled = _interopRequireDefault(require(Could not resolve "@emotion/styled"

    node_modules/@mui/styled-engine/index.js:9:21:
'@emotion/react' ;
        ╵                        ] Could not resolve "@emotion/react""@emotion/styled"X 

    node_modules/@mui/styled-engine/index.js:34:45:
));
         ╵                                              [ERROR]Could not resolve "@emotion/react"~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

X 

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

      34 │ export { ThemeContext, keyframes, css } from ~~~~~~~~~~~~~~~~~'@emotion/react'      9 │ import emStyled from 

    node_modules/@mui/styled-engine/GlobalStyles/GlobalStyles.js:3:23:
X [ERROR] Could not resolve "@emotion/react"      3 │ import { Global } from '@emotion/react'X [
                                                                                                                                                
  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will                                                 
  remove this error. You can also surround this "require" call with a try/catch block to handle this                                            
  failure at run-time instead of bundle-time.                                                                                                   
                                                                                                                                                
;
        ╵

    node_modules/@mui/styled-engine/GlobalStyles/GlobalStyles.js:3:23:
[      3 │ import { Global } from ERROR'@emotion/styled'Could not resolve "@emotion/react"X ~~~~~~~~~~~~~~~~;
         ╵                                              '@emotion/react'] ERROR
                                                                                                                                                
    node_modules/@mui/styled-engine/node/StyledEngineProvider/StyledEngineProvider.js:10:22:                                                    
      10 │ var _react2 = require("@emotion/react"[ERROR] Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/StyledEngineProvider/StyledEngineProvider.js:3:30:
      3 │ import { CacheProvider } from

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

;
        ╵                      Could not resolve "@emotion/react"]);
         ╵

    node_modules/@mui/styled-engine/StyledEngineProvider/StyledEngineProvider.js:3:30:
'@emotion/react'X ;
        ╵                        ~~~~~~~~~~~~~~~~                                                                                               

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

 ~~~~~~~~~~~~~~~~      3 │ import { CacheProvider } from Could not resolve "@emotion/react"

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

~~~~~~~~~~~~~~~~~'@emotion/react'

  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will
  remove this error.

;
        ╵

    node_modules/@mui/styled-engine/StyledEngineProvider/StyledEngineProvider.js:3:30:
~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

[ERROR~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

;
        ╵                               X [      3 │ import { CacheProvider } from ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

'@emotion/react'ERRORX []] Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/StyledEngineProvider/StyledEngineProvider.js:3:30:
      3 │ import { CacheProvider } from '@emotion/react';
        ╵                               ~~~~~~~~~~~~~~~~;
        ╵                               ~~~~~~~~~~~~~~~~X [
                                                                                                                                                
  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will                                                  
  remove this error.                                                                                                                            
                                                                                                                                                
X ERROR
                                                                                                                                                
  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will                                                  
  remove this error.                                                                                                                            
                                                                                                                                                
][ Could not resolve "@emotion/react"                                                                                                           

    node_modules/@mui/styled-engine/index.js:34:45:
X ERRORERRORCould not resolve "@emotion/react"]       34 │ export { ThemeContext, keyframes, css } from []

    node_modules/@mui/styled-engine/node/GlobalStyles/GlobalStyles.js:10:22:
'@emotion/react'Could not resolve "@emotion/styled"ERROR

    node_modules/@mui/styled-engine/node/index.js:46:45:
];
         ╵                                                    10 │ var _react2 = require(~~~~~~~~~~~~~~~~      46 │ var _styled = _interopRequireDefault(require( Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/node/index.js:47:21:
      47 │ var _react = require("@emotion/styled"Could not resolve "@emotion/react""@emotion/react"

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

"@emotion/react"X );
         ╵                                                                                                                                      

    node_modules/@mui/styled-engine/node/index.js:47:21:
      47 │ var _react = require("@emotion/react");
         ╵                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[);
         ╵                      ~~~~~~~~~~~~~~~~));
         ╵                                              ~~~~~~~~~~~~~~~~~ERROR
                                                                                                                                                
  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will                                                  
  remove this error. You can also surround this "require" call with a try/catch block to handle this                                            
  failure at run-time instead of bundle-time.                                                                                                   
                                                                                                                                                
                                                                                                                                                
                                                                                                                                                
  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will                                                  
  remove this error. You can also surround this "require" call with a try/catch block to handle this                                            
  failure at run-time instead of bundle-time.                                                                                                   
                                                                                                                                                
                                                                                                                                                
                                                                                                                                                
  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will                                                  
  remove this error. You can also surround this "require" call with a try/catch block to handle this                                            
  failure at run-time instead of bundle-time.                                                                                                   
                                                                                                                                                
                                                                                                                                                
                                                                                                                                                
  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will                                                 
  remove this error. You can also surround this "require" call with a try/catch block to handle this                                            
  failure at run-time instead of bundle-time.                                                                                                   
                                                                                                                                                
] Could not resolve "@emotion/styled"                                                                                                           

    node_modules/@mui/styled-engine/node/index.js:46:45:
X X [ERROR[]ERROR ]Could not resolve "@emotion/react" 

    node_modules/@mui/styled-engine/GlobalStyles/GlobalStyles.js:3:23:
Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/node/GlobalStyles/GlobalStyles.js:10:22:
      3 │ import { Global } from '@emotion/react'      10 │ var _react2 = require("@emotion/react");
         ╵                       ;
        ╵                        ~~~~~~~~~~~~~~~~X 

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

[X [ERRORERROR]]       46 │ var _styled = _interopRequireDefault(require( Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/node/index.js:47:21:
"@emotion/styled"X ));
         ╵                                              ~~~~~~~~~~~~~~~~~X [[ERRORCould not resolve "@emotion/react"

  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

ERROR] Could not resolve "@emotion/react"X []      47 │ var _react = require(

    node_modules/@mui/styled-engine/node/StyledEngineProvider/StyledEngineProvider.js:10:22:
ERROR

    node_modules/@mui/styled-engine/StyledEngineProvider/StyledEngineProvider.js:3:30:
      10 │ var _react2 = require("@emotion/react" "@emotion/react");
         ╵                      ]      3 │ import { CacheProvider } from Could not resolve "@emotion/styled"

    node_modules/@mui/styled-engine/node/index.js:46:45:
);
         ╵                        '@emotion/react';
        ╵                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

      46 │ var _styled = _interopRequireDefault(require(

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/node/index.js:47:21:
~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

      47 │ var _react = require("@emotion/styled"));
         ╵                                              "@emotion/react"~~~~~~~~~~~~~~~~~X [ERROR] Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/node/GlobalStyles/GlobalStyles.js:10:22:
      10 │ var _react2 = require("@emotion/react");
         ╵                       ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

X );
         ╵                      X

  You can mark the path "@emotion/styled" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

X [ERROR[ERROR]]  Could not resolve "@emotion/react"[~~~~~~~~~~~~~~~~

    node_modules/@mui/styled-engine/node/index.js:47:21:
ERROR

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

      47 │ var _react = require(]X "@emotion/react" Could not resolve "@emotion/react"Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/node/StyledEngineProvider/StyledEngineProvider.js:10:22:
[);
         ╵                      ~~~~~~~~~~~~~~~~

    node_modules/@mui/styled-engine/node/GlobalStyles/GlobalStyles.js:10:22:
ERROR

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

      10 │ var _react2 = require(]"@emotion/react" );
         ╵                       Could not resolve "@emotion/react"~~~~~~~~~~~~~~~~

    node_modules/@mui/styled-engine/StyledEngineProvider/StyledEngineProvider.js:3:30:


  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

      3 │ import { CacheProvider } from X '@emotion/react'[;
        ╵                               ERROR~~~~~~~~~~~~~~~~]

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

 X Could not resolve "@emotion/react"[

    node_modules/@mui/styled-engine/node/StyledEngineProvider/StyledEngineProvider.js:10:22:
ERROR      10 │ var _react2 = require(]"@emotion/react" );
         ╵                       Could not resolve "@emotion/react"~~~~~~~~~~~~~~~~

    node_modules/@mui/styled-engine/GlobalStyles/GlobalStyles.js:3:23:


  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

      3 │ import { Global } from X '@emotion/react'[;
        ╵                        ERROR~~~~~~~~~~~~~~~~]

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error.

 X Could not resolve "@emotion/react"[

    node_modules/@mui/styled-engine/node/GlobalStyles/GlobalStyles.js:10:22:
ERROR      10 │ var _react2 = require("@emotion/react"      10 │ var _react2 = require("@emotion/react");
         ╵                       );
         ╵                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

] Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/node/GlobalStyles/GlobalStyles.js:10:22:
      10 │ var _react2 = require("@emotion/react");
         ╵                       ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

X [ERROR] 

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/node/StyledEngineProvider/StyledEngineProvider.js:10:22:
      10 │ var _react2 = require("@emotion/react");
         ╵                       X [ERROR] Could not resolve "@emotion/react"

    node_modules/@mui/styled-engine/node/StyledEngineProvider/StyledEngineProvider.js:10:22:
      10 │ var _react2 = require("@emotion/react");
         ╵                       ~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

~~~~~~~~~~~~~~~~

  You can mark the path "@emotion/react" as external to exclude it from the bundle, which will
  remove this error. You can also surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.

src/webview/openshift-terminal/app/terminalMultiplexer.tsx(6,31): error TS2307: Cannot find module '@emotion/react' or its corresponding type declarations.
```

**Review by msivasubramaniaan — CHANGES_REQUESTED** [review]

File: src/webview/openshift-terminal/app/terminalMultiplexer.tsx
Comment (datho7561) [reply src/webview/openshift-terminal/app/terminalMultiplexer.tsx]: Good catch! I replaced that import with the correct import. However, `@emotion/react` should be a transitive dependency of Material UI, so I don't know why you are getting that error. Let me know if it's working after the change.

Comment (msivasubramaniaan) [reply src/webview/openshift-terminal/app/terminalMultiplexer.tsx]: Yes still the issue exists. `@mui/styled-engine` internally referred `@emotion/styled` and `@emotion/react` 

image

Comment (datho7561) [reply src/webview/openshift-terminal/app/terminalMultiplexer.tsx]: @msivasubramaniaan were you able to fix the dependency issues? I wasn't able to replicate them on my computer

**Comment by datho7561** [comment]
@JessicaJHee @rgrunber let me know if you think it's good. I'll have to address the merge conflict before we can merge the PR.

**Review by JessicaJHee — APPROVED** [review]
LGTM!

File: src/webview/openshift-terminal/app/terminalMultiplexer.tsx
Comment (datho7561) [reply src/webview/openshift-terminal/app/terminalMultiplexer.tsx]: @msivasubramaniaan do you still run into this issue? if not, I would like to merge this PR

Comment (msivasubramaniaan) [reply src/webview/openshift-terminal/app/terminalMultiplexer.tsx]: Sorry for late reply as busy with serverless function. Will review today.

**Review by msivasubramaniaan — APPROVED** [review]
LGTM

**Pull request merged**
